### PR TITLE
rock and stone

### PR DESCRIPTION
**Pull request by AwkwardDryad**
[FEATURE]

## About the PR
- Dwarves have a 3% chance to fart out a glowstick (Reference to Deep Rock Galactic flares)

## Why's this needed?
- ........reasons.........

### CODE REVIEW

File: code/mob/living/carbon/human/procs/emote.dm [hunk 1]
```diff
@@ -230,6 +230,18 @@
 		#ifdef DATALOGGER
 						game_stats.Increment("farts")
 		#endif
+				if(src.mutantrace && src.mutantrace.name == "dwarf" && prob(3))
+					var/glowsticktype = pick(/obj/item/device/light/glowstick,
+					/obj/item/device/light/glowstick/white,/obj/item/device/light/glowstick/yellow,
+					/obj/item/device/light/glowstick/blue,/obj/item/device/light/glowstick/purple,
+					/obj/item/device/light/glowstick/pink,/obj/item/device/light/glowstick/cyan,
+					/obj/item/device/light/glowstick/orange,/obj/item/device/light/glowstick/red)
```
Comment (ZeWaka) [inline code/mob/living/carbon/human/procs/emote.dm:238]: why is this a manual list and not just typesof()

**Comment by glassofmilk** [comment]
To the bone!

File: code/mob/living/carbon/human/procs/emote.dm (hunk 1 above)
Comment (AwkwardDryad) [reply code/mob/living/carbon/human/procs/emote.dm:238]: that.........exists? :D

**Comment by Tenryumi** [comment]
b49cfd488de4fd2dcf352a703cb5bfca7c6fe56e7afdfc6d1cd86dc3005474a2

yes

good

**Comment by Avatharian** [comment]
FOR KARL!

File: code/mob/living/carbon/human/procs/emote.dm [hunk 2]
```diff
@@ -230,6 +230,17 @@
 		#ifdef DATALOGGER
 						game_stats.Increment("farts")
 		#endif
+				if(src.mutantrace && src.mutantrace.name == "dwarf" && prob(3))
+					var/list/glowsticktypes = list()
+					for(var/type in typesof(/obj/item/device/light/glowstick))
+						glowsticktypes += type
+					var/glowsticktype = pick(glowsticktypes)
+					var/obj/item/device/light/glowstick/G = new glowsticktype
```
Comment (ZeWaka) [inline code/mob/living/carbon/human/procs/emote.dm:238]:
```suggestion
					var/obj/item/device/light/glowstick/G = new pick(typesof(/obj/item/device/light/glowstick))
```

Comment (AwkwardDryad) [reply code/mob/living/carbon/human/procs/emote.dm]: There was a bit of an issue with the compiler, but it's fixed in the latest commit :heart: :tada:

**Comment by Adharainspace** [comment]
nt farms dwarves for glowsticks

**Comment by Kyle2143** [comment]
Not to be a downer, but I don't really understand this joke and don't see why it would be funny to add.

**Comment by AwkwardDryad** [comment]
> Not to be a downer, but I don't really understand this joke and don't see why it would be funny to add.

It's a slight reference to deep rock galactic, a game in which a team of four players play as space dwarves mining out a hostile planet. But the patch references the fact that the f key in deep rock throws flares from seemingly out of nowhere. Therefore F key + flare manifested from thin air -> dwarf producing flare in body -> F key to fart glowstick in space station 13.

**Comment by Avatharian** [comment]
Quite a few of the most commonly used items in Goonstation are references to other stuff. [Cyalume Saber](https://sites.google.com/site/steinsgatepokemongym/_/rsrc/1433712531278/home/pokemon/future-gadget-6/Futuregadget06_02.png?height=225&width=400), anyone? I see no reason why this couldn't be added as a fun easter-egg as well.

**Comment by Kyle2143** [comment]
Yes this is good, forgot about this pr. I asked because I didn't know that this was even a reference to anything specific, the pr description didn't give any clues and I assumed that it was some joke about how dwarves typically mine underground and would therefore need light to see. Which doesn't really track but was all I could go on. 

But in the future if someone wants to add a joke that is purely a reference to something, they should give a description in the PR so we don't have to ask. Not everybody is is familiar with every game.

**Pull request merged**